creating_citation_counts_tsv/scripts/summarize_gene_citations_all_species.py

```diff
@@ -0,0 +1,335 @@
+    gene_counts = gene2pubmed.join(year_pmid).map(lambda x: (x[1][0], 1)).reduceByKey(add).join(gene2pubmed_to_year).map(lambda x: ( str(x[0]), (x[1][1] ,x[1][0])))
+    
+    # Outputing the first element in the gene_counts variable
+    print(gene_counts.take(1))
+    
+    # Returning the gene_counts variable
+    return gene_counts
+
+def getting_gene_info(species, gene_counts):
+    """ 
+    Getting a list of the gene information per gene and species' taxonomy name.
+    """
+    
+    # Adding gene's information to the list created from getting_recent_gene2pubmed per gene and saving it to the gene_info variable
+    gene_info = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/gene_info")
+                    .filter(lambda x: x[0] != '#')
+                    .map(lambda x: x.split('\t'))
+                    .map(lambda x: (str(x[1]), (str(x[0]), str(x[2]), str(x[6]), str(x[8]))))).join(gene_counts).map(lambda x: (x[1][0][1].upper(), (x[0], x[1][0][2], x[1][0][3], x[1][1][1], x[1][0][0])))
+    
+    # Outputing the first element in the gene_info variable
+    print(gene_info.take(1))
+    
+    # Getting the taxonomy ID for the species and saving it to the tax_id variable
+    tax_id = str(gene_info.collect()[0][1][4])
+    
+    # Outputing the first element in the tax_id variable
+    print(tax_id)
+
+    # Getting the taxonomy information for the taxonomy ID given and saving it to the taxonomy variable
+    taxonomy = (sc.textFile("creating_citation_counts_tsv/taxonomy_name")
+                        .filter(lambda x: x[0] != '#')
+                        .map(lambda x: x.split('\t'))
+                        .filter(lambda x: int(tax_id) == int(x[0]) )
+                        .filter(lambda x: "scientific name" in x[6] )
+                        .map(lambda x: ((str(x[0]), str(x[2]), str(x[6]))))
+                     )
+    
+    # Getting the taxonomy name from the taxonomy information and saving it to the tax_name variable
+    tax_name = taxonomy.collect()[0][1]
+    
+    # Outputing the tax_name variable
+    print(tax_name)
+
+    # Returning the gene_info and tax_name variable
+    return gene_info, tax_name
+
+def getting_refGene(species, gene_info): 
+    """ 
+    Getting a list of the gene reference information per gene.
+    """
+    
+    # Going through all the files in the species directory 
+    for refGene_file in os.listdir(f"creating_citation_counts_tsv/data/{species}"):
+         #Since the reference file is uniquely named and could not be hard-coded, this will filter out "gene_info" and "gene2pubmed" to read the reference file
+        if refGene_file != "gene_info" and "gene2pubmed":
+            # Adding gene's reference information to the list created from getting_gene_info per gene and saving it to the refGene variable
+            refGene = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/{refGene_file}")
```

Comment:
This is difficult to read, even with some separation over lines.  

Consider breaking up this very complex assignment into multiple smaller assignments, building up into the assignment of `ref_gene`.

Reply:
I scared it might be a lot of memory in RAM, if make separate variables